java_runtime/src/classes/java/util/vector.rs

```diff
@@ -177,6 +180,59 @@ impl Vector {
         Ok(())
     }
 
+    async fn last_index_of(jvm: &Jvm, _: &mut RuntimeContext, this: ClassInstanceRef<Self>, element: ClassInstanceRef<Object>) -> Result<i32> {
+        tracing::debug!("java.util.Vector::lastIndexOf({:?}, {:?})", &this, &element);
+
+        let rust_vector = Self::get_rust_vector(jvm, &this).await?;
+        let index = rust_vector.lock().await.len() - 1;
+
+        let index: i32 = jvm
+            .invoke_virtual(&this, "lastIndexOf", "(Ljava/lang/Object;I)I", (element, index as i32))
+            .await?;
+
+        Ok(index)
+    }
+
+    async fn last_index_of_index(
+        jvm: &Jvm,
+        _: &mut RuntimeContext,
+        this: ClassInstanceRef<Self>,
+        element: ClassInstanceRef<Object>,
+        index: i32,
+    ) -> Result<i32> {
+        tracing::debug!("java.util.Vector::lastIndexOf({:?}, {:?}, {:?})", &this, &element, index);
+
+        let rust_vector = Self::get_rust_vector(jvm, &this).await?;
+        let size = rust_vector.lock().await.len();
+
+        if index as usize >= size {
+            return Err(jvm
+                .exception("java/lang/IndexOutOfBoundsException", &format!("{} >= {}", index, size))
+                .await);
+        }
+
+        if element.is_null() {
+            for i in (0..=index as usize).rev() {
+                let e: ClassInstanceRef<Object> = rust_vector.lock().await.get(i).unwrap().clone();
+                if e.is_null() {
+                    return Ok(i as i32);
+                }
+            }
+        } else {
+            let element: Box<dyn ClassInstance> = element.into();
+
+            for i in (0..=index as usize).rev() {
+                let e: Box<dyn ClassInstance> = rust_vector.lock().await.get(i).unwrap().clone().into();
+
+                if e.equals(&*element)? {
+                    return Ok(i as i32);
+                }
+            }
+        }
```

Comment:
```suggestion
        let vector = rust_vector.lock().await;
        let size = vector.len();

        if index as usize >= size {
            return Err(jvm
                .exception("java/lang/IndexOutOfBoundsException", &format!("{} >= {}", index, size))
                .await);
        }
        
        for item in vector.iter().take(i).rev() {
        ...
```

Per [spec](https://docs.oracle.com/javase/8/docs/api/index.html?java/util/Vector.html), all `Vector` operation should be synchronized